File: archLintRulesTestDemo/src/main/java/me/ycdev/android/arch/demo/activity/LintGoodActivity.java
package me.ycdev.android.arch.demo.activity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import me.ycdev.android.arch.activity.AppCompatBaseActivity;
import me.ycdev.android.arch.demo.R;


public class LintGoodActivity extends AppCompatBaseActivity { // lint good

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lint_good);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_lint_good, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: archLintRulesTestDemo/src/main/java/me/ycdev/android/arch/demo/activity/LintGoodActivity.kt
package me.ycdev.android.arch.demo.activity

import android.view.Menu
import android.view.MenuItem
import me.ycdev.android.arch.activity.AppCompatBaseActivity

class LintGoodActivity : AppCompatBaseActivity() { // lint good

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        item.itemId

        return super.onOptionsItemSelected(item)
    }
}
